contrib/config/consul/consul.go

```diff
@@ -0,0 +1,166 @@
+// Copyright GoFrame Author(https://goframe.org). All Rights Reserved.
+//
+// This Source Code Form is subject to the terms of the MIT License.
+// If a copy of the MIT was not distributed with this file,
+// You can obtain one at https://github.com/gogf/gf.
+
+// Package consul implements gcfg.Adapter using consul service.
+package consul
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/gogf/gf/v2/encoding/gjson"
+	"github.com/gogf/gf/v2/errors/gerror"
+	"github.com/gogf/gf/v2/frame/g"
+	"github.com/gogf/gf/v2/os/gcfg"
+	"github.com/hashicorp/consul/api"
+	"github.com/hashicorp/consul/api/watch"
+)
+
+const LoggerName = "consul-adapter"
+
+// Config is the configuration object for consul client.
+type Config struct {
+	Params api.Config `v:"required"`
```

Comment:
Is it better using `ConsulConfig` instead of `Params` for this attribute name?

Reply:
Yes